_posts/2019-02-18-event-store-5.0.0-release.md

```diff
@@ -0,0 +1,193 @@
+All events having a `$correlationId` property (configurable in the `$by_correlation_id` projection source) will be emitted to a stream called `$bc-<correlation id>`
+* [#1678](https://github.com/EventStore/EventStore/pull/1678) - **(Server)** Structured Logging  
+By default, EventStore logs and stats will now be saved in the JSON format. Each line written to the logs or stats file is a serialized JSON object. The main benefit of this new feature is that you can easily ingest your logs and stats into your log aggregation or data analysis platform without requiring any grok filter/custom code.
+* [#1842](https://github.com/EventStore/EventStore/pull/1842) - **(Server)** Add a maximum index level for automatically merging PTables. This option is useful in situations where large index merges are slowing down writes due to heavy disk usage. With this option, large PTables can be merged during off-peak hours.
+  * A new configuration option named `MaxAutoMergeIndexLevel (int)` has been added which defines the highest PTable level up to which PTable index files will be merged automatically. Let's denote the maximum merge level you have set in your configuration by `L` to simplify the explanation below.
+  * To merge PTables that have reached the specified level `L`, an endpoint URL can be manually triggered by doing a `POST` request to `/admin/mergeindexes`. This will merge all the PTables at level `L` to level `L+1`. Also, if there are any PTables that are at level `L+2` or higher they will be merged down to level `L+1`.
+  * Once set, the maximum merge level can be decreased in your configuration. However, increasing it in your configuration will require a full index rebuild.  
+  * For quick reference, default PTable levels are as follows (assuming that you have not changed `MaxMemTableSize` which is `1M` by default):  
+    * Level 0 - 1M index entries
+    * Level 1 - 2M index entries
+    * Level 2 - 4M index entries
+    * Level 3 - 8M index entries
+    * Level 4 - 16M index entries
+    * Level 5 - 32M index entries  
+    --- Depending on your hardware configuration, merges can take several minutes as from this level, you can verify your logs to get a better idea of the amount of time taken ---
+    * Level 6 - 64M index entries  
+    * Level 7 - 128M index entries  
+    * Level 8 - 256M index entries  
+    ...
+    * Level K - 2^K x 1M index entries
+* [#1774](https://github.com/EventStore/EventStore/pull/1774) - **(Projections)** Add a `FaultOutOfOrderProjections` option (default: `True`) to allow the user to specify if a projection should be faulted when there is a discontinuity in event ordering. For example, if a stream has a `$maxAge` or `$maxCount`, a projection processing this stream may not receive some events if they have already "expired". In case `FaultOutOfOrderProjections` is set to `False`, an error will simply be logged and the projection will continue processing events.
```

Comment:
```suggestion
* [#1774](https://github.com/EventStore/EventStore/pull/1774) - **(Projections)** Add a `FaultOutOfOrderProjections` option (default: `True`) to allow the user to specify if a projection should be faulted when there is a discontinuity in event ordering. For example, if a stream has a `$maxAge` or `$maxCount`, a projection processing this stream may not receive some events if they have already "expired". In case `FaultOutOfOrderProjections` is set to `False`, an error is logged and the projection will continue processing events.
```